src/UI/Layout.hs

```diff
@@ -62,12 +79,22 @@ fitLayoutAndAnnotate :: Real a => Size (Maybe a) -> Layout a (Size a) -> ALayout
 fitLayoutAndAnnotate = fitLayoutWith (annotatingBidi layoutAlgebra)
 
 layoutAlgebra :: Real a => Algebra (Fitting (LayoutF a) a) (Maybe (Rect a))
-layoutAlgebra (Cofree (offset, maxSize) runC layout) = case layout of
-  Pure size | maxSize `encloses` size -> Just (Rect offset (fromMaybe <$> size <*> maxSize))
+layoutAlgebra (Cofree (alignment, offset, maxSize) runC layout) = case layout of
+  Pure size | maxSize `encloses` size -> Just $ case alignment of
+    Leading -> Rect offset minSize
+    Trailing -> Rect offset { x = x offset + widthDiff} minSize
+    Centre -> Rect offset { x = x offset + fromIntegral (widthDiff `div'` 2 :: Int)} minSize
+    Full -> Rect offset fullSize
+    where minSize = fullSize { width = width size }
+          fullSize = fromMaybe <$> size <*> maxSize
+          widthDiff = maybe 0 (+ negate (width size)) (width maxSize)
   Free runF layout -> case layout of
     Inset by child -> Rect offset . (2 * by +) . size <$> runC (runF child)
     Offset by child -> Rect offset . (pointSize by +) . size <$> runC (runF child)
     GetMaxSize -> runC (runF maxSize)
+    Align _ child -> do
+      Rect _ size <- runC (runF child)
+      pure $ Rect offset (fromMaybe <$> size <*> maxSize)
```

Comment:
`Align` is the only branch term whose size is not determined solely by its child. That is, an `Align` term may be sized a variable amount larger than its child. E.g.:

```haskell
fitLayout (Size (Just 50) Nothing) alignLeft (pure (Size 10 10))
```

will result in a layout like so:

```
| … 40 … |10||
```